pkg/tracing/config/tracing.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	ConfigName = "config-tracing"
+
+	enableKey                  = "enable"
+	zipkinCollectorEndpointKey = "zipkin-collector-endpoint"
+	debugKey                   = "debug"
+	sampleRateKey              = "sample-rate"
+)
+
+type Config struct {
+	Enable      bool
+	EndpointURL string
+	Debug       bool
+	SampleRate  float64
+}
+
+func (cfg *Config) Equals(other *Config) bool {
+	return other.Enable == cfg.Enable && other.EndpointURL == cfg.EndpointURL && other.Debug == cfg.Debug && other.SampleRate == cfg.SampleRate
+}
+
+func NewTracingConfigFromMap(cfgMap map[string]string) (*Config, error) {
+	tc := Config{}
```

Comment:
Can I recommend to setup `tc` with default values, e.g.:
```
tc := &Config{
Enable: false,
...,
}
```
and then only set the overrides that are present in the configmap.
That surely is going to be less code and more readable.